### ISSUE TITLE
Contact Sensor listed as "mcs" category doesn't show up in HA.

### ISSUE THREAD

**Issue by strasharo** [tuya/tuya-home-assistant#37]
Hello. I'm running v 1.3 of the integration with an up to date HASS and my "mcs" category door sensor doesn't show up in HASS.
As far as I understand from the table here https://github.com/tuya/tuya-home-assistant/wiki/Supported-Device-Category mcs is a supported category, is that right or I'm doing something wrong? It's linked in the project and works fine in the Smart Life app.

**Comment by Gferretta** [comment]
Same here

I add a Door Window sensor to TUYA app and work ok
In Tuya API linked accounts and works too.

But in HA the sensor "mcs" not appear.

Only one virtual device (RGB Bulb) that I add in TUYA cloud project for testing only.   If I remove it, integration in HA appear empty.

This is result of API Explorer on tuya platform for "Door Window Sensor device":

> 

{
  "result": {
    "active_time": 1625156459,
    "biz_type": 0,
    "category": "mcs",
    "create_time": 1625147323,
    "icon": "smart/icon/ay1525749833414yotNt/92722d1b64636ed1dfe3107187643324.jpg",
    "id": ".....................1a8.......",
    "ip": "181.0.XX.YYY",
    "lat": "",
    "local_key": ".........18...........",
    "lon": "",
    "name": "Puerta",
    "online": true,
    "owner_id": "..............7",
    "product_id": ".....................54b9.....",
    "product_name": "Contact Sensor",
    "status": [
      {
        "code": "doorcontact_state",
        "value": false
      }
    ],
    "sub": false,
    "time_zone": "-03:00",
    "uid": "...............",
    "update_time": 1625156835,
    "uuid": ".....................1a8......."
  },
  "success": true,
  "t": 1625158456875
}

The RGB Bulb change immediately between app Tuya and HA..... work perfectly OK, but is virtual. :(

![Tuya v2 HA](https://user-images.githubusercontent.com/65413059/124166431-34671180-da79-11eb-94ce-8bfd40d4f3f2.jpg)

**Comment by NoLooseEnds** [comment]
Same for me. I have the contact sensors in the Tuya app, but they do not show in Home Assistant.

**Comment by strasharo** [comment]
Here's mine:

```
{
  "result": {
    "active_time": 1624891087,
    "biz_type": 0,
    "category": "mcs",
    "create_time": 1624887351,
    "icon": "smart/icon/ay15427647462366edzT/042f961da388104417809668b6c83e2b.jpg",
    "id": "<scrubbed>",
    "ip": "<scrubbed>",
    "lat": "",
    "local_key": "<scrubbed>",
    "lon": "",
    "name": "Server room",
    "online": true,
    "owner_id": "<scrubbed>",
    "product_id": "3slpirlnlqp0bqo1",
    "product_name": "Contact Sensor",
    "status": [
      {
        "code": "doorcontact_state",
        "value": false
      },
      {
        "code": "battery_state",
        "value": "high"
      }
    ],
    "sub": false,
    "time_zone": "+02:00",
    "uid": "<scrubbed>",
    "update_time": 1625165406,
    "uuid": "<scrubbed>"
  },
  "success": true,
  "t": 1625165411107
}
```

**Comment by Gferretta** [comment]
> Here's mine:
> The device appears to be like this one:
> https://templates.blakadder.com/TY01.html
> 
> ```
> {
>   "result": {
>     "active_time": 1624891087,
>     "biz_type": 0,
>     "category": "mcs",
>     "create_time": 1624887351,
>     "icon": "smart/icon/ay15427647462366edzT/042f961da388104417809668b6c83e2b.jpg",
>     "id": "<scrubbed>",
>     "ip": "<scrubbed>",
>     "lat": "",
>     "local_key": "<scrubbed>",
>     "lon": "",
>     "name": "Server room",
>     "online": true,
>     "owner_id": "<scrubbed>",
>     "product_id": "3slpirlnlqp0bqo1",
>     "product_name": "Contact Sensor",
>     "status": [
>       {
>         "code": "doorcontact_state",
>         "value": false
>       },
>       {
>         "code": "battery_state",
>         "value": "high"
>       }
>     ],
>     "sub": false,
>     "time_zone": "+02:00",
>     "uid": "<scrubbed>",
>     "update_time": 1625165406,
>     "uuid": "<scrubbed>"
>   },
>   "success": true,
>   "t": 1625165411107
> }
> ```


Hi
My sensor not return battery state.
Why?
What can do?

**Comment by strasharo** [comment]
@tsutsuku Can I assist somehow with the troubleshooting, is there any other info that I can provide? I've checked the logs but didn't notice any related exception, everything appears to be working fine, the device just doesn't show up.

**Comment by mrwhite31** [comment]
I am having the same problem. Category mcs only displaying the battery state. Looking into the logger integration it is calling the "Get device specification attributes" API /v1.0/devices/{device_id}/specifications. In the JSON response the boolean value of doorcontact_state and temper_alarm are returned with "values": "{}" independent of the effective door state. Maybe the root cause of this issue?

Response of /v1.0/devices/{device_id}/specifications
{
  "result": {
    "category": "mcs",
    "functions": [],
    "status": [
      {
        "code": "doorcontact_state",
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "battery_percentage",
        "type": "Integer",
        "values": "{\"unit\":\"%\",\"min\":0,\"max\":100,\"scale\":0,\"step\":1}"
      },
      {
        "code": "temper_alarm",
        "type": "Boolean",
        "values": "{}"
      }
    ]
  },
  "success": true,
  "t": 1................23.45
}

**Comment by mrwhite31** [comment]
Quick fix for me was to change the value on line 47 of binary_sensor.py from "switch" to "doorcontact_state".
My doorsensors are now appearing in Homeassistant

**Comment by NoLooseEnds** [comment]
@mrwhite31 I tried doing that, but no go here. 

Restarted HA, but the Tuya integration does not show my two door sensors (they show up in Tuya Cloud, and I get a similar API response as the ones above).

Did you do anything else?

**Comment by mrwhite31** [comment]
I've reloaded the Tuya integration and restarted HA after updating the file. Works for me. Does the response of the specification APo return "doorcontact_sensor for your device? And is it category "mcs"?

**Comment by NoLooseEnds** [comment]
Actually, it seems like the category is ´mc´

```
{
  "result": {
    "active_time": 1599729004,
    "biz_type": 0,
    "category": "mc",
    "create_time": 1599729004,
    "icon": "smart/icon/1528364671b3tgndtkqr_0.png",
    "id": "<removed>",
    "ip": "<removed>",
    "lat": "<removed>",
    "local_key": "<removed>",
    "lon": "<removed>,
    "name": "Stair gate",
    "online": true,
    "owner_id": "<removed>",
    "product_id": "oSQljE9YDqwCwTUA",
    "product_name": "Door Sensor",
    "status": [
      {
        "code": "doorcontact_state",
        "value": true
      },
      {
        "code": "battery_state",
        "value": "high"
      }
    ],
    "sub": false,
    "time_zone": "+01:00",
    "uid": "<removed>",
    "update_time": 1625297596,
    "uuid": "<removed>"
  },
  "success": true,
  "t": 1625312600945
}
```

**Comment by NoLooseEnds** [comment]
I got it to show up by adding

`"mc",  # Door Window Sensor` to the TUYA_SUPPORT_TYPE array:

```
TUYA_SUPPORT_TYPE = [
    "mc",  # Door Window Sensor
    "mcs",  # Door Window Sensor
    "ywbj",  # Smoke Detector
    "rqbj",  # Gas Detector
    "pir",  # PIR Detector
    "sj",  # Water Detector
    "sos",  # Emergency Button
    "hps",  # Human Presence Sensor
]
```

For reference – the sensors I got is [NEO COOLCAM Wifi Door Window Sensor ](https://www.aliexpress.com/item/32858144085.html)


Restarted HA again, and I got my door sensors.

**Comment by Gferretta** [comment]
> Quick fix for me was to change the value on line 47 of binary_sensor.py from "switch" to "doorcontact_state".
> My doorsensors are now appearing in Homeassistant

This hack work for me.
Thanks
Waiting for official release.

The only problem is that it doesn't show any value for the battery! ; (
I don't know if that depends on the firmware of the sensor or what other reason.

![Tuya v2 HA 1](https://user-images.githubusercontent.com/65413059/124355341-fbe54600-dbe6-11eb-9342-83dc458e04d9.jpg)

**Comment by NoLooseEnds** [comment]
I don't have any values for battery in HA either, but I can see the `battery_state` in the [API response](https://github.com/tuya/tuya-home-assistant/issues/37#issuecomment-873395138).

**Comment by strasharo** [comment]
> Quick fix for me was to change the value on line 47 of binary_sensor.py from "switch" to "doorcontact_state".
> My doorsensors are now appearing in Homeassistant

That fixed it for me! Thanks a lot!

**Comment by mrwhite31** [comment]
I do have three different types of contact sensors. Only one of them showing the battery status. Looking into the API responses I see three different types of battery states. Only one returned correctly is "battery_percentage". Then there is also "battery_value" returning an integer and "battery_state" which can me "high|low|mid". In sensor.py only "battery_percentage" is in.

my approach here would be to add these states and map the values to a percentage.

**Comment by tsutsuku** [comment]
@Gferretta  mcs fixed in v1.3.1.

**Comment by Gferretta** [comment]
> @Gferretta mcs fixed in v1.3.1.

Updated!
Thank a lot

Yet not info of battery!  ;(

    "product_id": "**ikfjwqdbpnh154b9**",
    "product_name": "Contact Sensor",
    "status": [
      {
        "code": "doorcontact_state",
        "value": false
      }
    ],
    "sub": false,


I found that [Wi-Fi Door Sensor](https://expo.tuya.com/product/575579) with same "product_id" code.... maybe help with the battery info.

Thank again

**Comment by strasharo** [comment]
Sensor shows up fine in v1.3.1. Battery status isn't displayed though, so I'll open a separate defect for it.
Thanks a lot @tsutsuku!
